docs/specification.md

```diff
@@ -1,153 +1,207 @@
 
-- result: [`SignedMEVPayloadHeader`](#signedmevpayloadheader)
-- error: code and message set in case an exception happens while getting the payload.
+As `compute_signing_root` takes `SSZObject` as input, client software should convert in-protocol messages to their SSZ representation to compute the signing root and Builder API messages to the SSZ representations defined [above](#sszobjects).
+
+## Methods
 
-### relay_proposeBlindedBlockV1
+### `builder_setFeeRecipientV1`
 
 #### Request
 
-- method: `relay_proposeBlindedBlockV1`
+- method: `builder_setFeeRecipientV1`
 - params:
-  1. [`SignedBlindedBeaconBlock`](#signedblindedbeaconblock)
-  2. [`SignedMEVPayloadHeader`](#signedmevpayloadheader)
+  1. `feeRecipient`: `DATA`, 20 Bytes - Address of account which should receive fees.
+  2. `timestamp`: `QUANTITY`, uint64 - Unix timestamp of announcement.
+  2. `publicKey`: `DATA`, 48 Bytes - Public key of validator.
+  3. `signature`: `DATA`, 96 Bytes - Signature over `feeRecipient`.
 
 #### Response
 
-- result: [`ExecutionPayloadV1`](https://github.com/ethereum/consensus-specs/blob/v1.1.10/specs/bellatrix/beacon-chain.md#executionpayload)
-- error: code and message set in case an exception happens while proposing the payload.
-
-Technically, this call only needs to return the `transactions` field of [`ExecutionPayloadV1`](https://github.com/ethereum/consensus-specs/blob/v1.1.10/specs/bellatrix/beacon-chain.md#executionpayload), but we return the full payload for simplicity.
-
-### Types
-
-#### SignedMEVPayloadHeader
+- result: `null`
+- error: code and message set in case an exception happens while getting the payload.
 
-See [here](https://github.com/ethereum/consensus-specs/blob/v1.1.10/specs/phase0/beacon-chain.md#custom-types) for the definition of fields like `BLSSignature`
+#### Specification
+1. Builder software **MUST** verify `signature` is valid under `publicKey`.
+2. Builder software **MUST** respond to requests where `timestamp` is before the latest announcement from the validator or more than +/- 10 seconds from the current time with `-32006: Invalid timestamp`.
+3. Builder software **MUST** store `feeRecipient` in a map keyed by `publicKey`.
 
-- message: [MEVPayloadHeader](#mevpayloadheader)
-- signature: BLSSignature
+### `builder_getHeaderV1`
 
-#### MEVPayloadHeader
+#### Request
 
-- payloadHeader: [`ExecutionPayloadHeaderV1`](https://github.com/ethereum/consensus-specs/blob/v1.1.10/specs/bellatrix/beacon-chain.md#executionpayloadheader)
-- feeRecipientDiff: Quantity, 256 Bits - the change in balance of the feeRecipient address
+- method: `builder_getHeaderV1`
+- params:
+  1. `hash`: `DATA`, 32 Bytes - Hash of block which the validator intends to use as the parent for its proposal.
```

Comment:
should the validator public key be part of the request params or else how does the builder return `Unknown validator` if the validator public key has not been mapped to a `feeRecipient`?

Reply:
My expectation was that the builder would know what validator is planning to propose for the current slot and would return `Unknown validator` if it had not received a `setFeeRecipient` message from the validator. I will update the wording here, because "recovered" doesn't make much sense in this context.

This does make me wonder if the builder should also return the slot the slot it is building for in case there is a mismatch?

Reply:
There could skip slots on top of `hash`
```
hash -> request/propose
hash -> skipped_slot -> request/propoase
hash -> skipped_slot -> skipped_slot -> request/propose
```
The builder may utilize the current wall block but if there's a tiny clock synchronization mismatch between the BN and builder then it may fall apart. It may be safer to include an additional `slot` or `proposer_index` or `pub_key` in the request as an additional anchor point

Reply:
@terencechain thanks for the brainstorming on this. the PR is now merged, please open an issue/PR to continue the thread, if applicable 🙏